### DISCUSSION TITLE
First database query after startup runs very slowly

### DISCUSSION THREAD

**Discussion by PeterGoedeke** [prisma/prisma#9695] [Q&A]
I'm having an issue where the first prisma query after starting up the app will take much longer than subsequent queries (e.g. 900ms compared to 40ms). Whether the first query is the same as the following queries or different, there is still an extremely noticeable speedup between the first and second query of at least an order of magnitude every time the app is started up. This has been reproduced with two separate node apps connecting to two separate databases. It also occurs when attempting to connect to a local database with the first query taking about 120ms and the second taking around 7ms. When connecting to the same database using the Python SQLAlchemy ORM, this slowdown on the first query is not nearly as noticeable. The databases tested against were both running Postgres. Thanks.

**Comment by ryands17** [comment]
@PeterGoedeke 👋 

I'm pretty sure this is due to Prisma connecting to the database on the first request at the very first startup of the app. Whenever you query for something, Prisma lazily connects to the database so you might find that slower compared to other requests. Have a look at [this doc](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/connection-management#connect) for more info.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
